pages/profile/[name]/[[...tab]]/index.tsx

```diff
@@ -172,6 +179,71 @@ export async function getServerSideProps(context: GetServerSidePropsContext) {
     profilePageProps.totalRows = query.totalRows;
   }
 
+  if (profileTab === ProfileTab.Profile) {
+    const levelsCompletedByDifficultyData = await StatModel.aggregate([
+      {
+        $match: {
+          $and: [ {userId: user._id }, { complete: true}],
+        },
+      },
+      {
+        $lookup: {
+          from: 'levels',
+          localField: 'levelId',
+          foreignField: '_id',
+          as: 'levelInfo',
+        },
+      },
+      {
+        $unwind:"$levelInfo"
+      },
+      {
+        $project: {
+          difficulty: "$levelInfo.calc_difficulty_estimate"
+        }
+      }
+    ]);
+
+    let levelsCompletedByDifficulty:levelsCompletedByDifficultyInterface = {};
+    levelsCompletedByDifficulty["Pending"] = 0;
+    const difficultyList = getDifficultyList();
```

Comment:
Feels kind of off that Pending is not included in the getDifficultyList()... Maybe Pending shouldn't be returned at all?

Reply:
Although you can filter by pending levels on the search page, so maybe it's worth having it as a category? Can see doing either way.